lib/services/search.js

```diff
@@ -0,0 +1,404 @@
+'use strict';
+
+const elastic = require('elasticsearch'),
+  bluebird = require('bluebird'),
+  _ = require('lodash'),
+  util = require('util'),
+  log = require('./log').withStandardPrefix(__filename),
+  endpoint = process.env.ELASTIC_HOST || '',
+  serverConfig = {
+    host: endpoint,
+    maxSockets: 500,
+    apiVersion: '1.7',
+    // log: ['error', 'trace'],
+    defer: require('../utils/defer')
+  };
+
+var client,
+  aliasSuffix = '';
+
+/**
+ * Test whether or not the client is connected
+ * to Elastic
+ *
+ * @param  {object} currentClient
+ * @returns {Promise}
+ */
+function healthCheck(currentClient) {
+  return currentClient.ping({
+    requestTimeout: 1000
+  }).then(function () {
+    log('info', 'Elasticsearch cluster is up!');
+    return bluebird.resolve();
+  }).catch(function (error) {
+    log('info', 'Elasticsearch cluster is down!');
+    return bluebird.reject(error);
+  });
+}
+
+/**
+ * Create an Elasticsearch index
+ *
+ * @param {string} index
+ * @param {object} mappings
+ * @returns {Promise}
+ */
+function initIndex(index, mappings) {
+  return client.indices.create({
+    index: index,
+    body: mappings
+  }).then(function () {
+    log('info', 'Successfully created ' + index);
+  })
+    .catch(function (error) {
+      log('error', error);
+      return bluebird.reject(error);
+    });
+}
+
+/**
+ * Check if an Elasticsearch index exists
+ *
+ * @param {string} index
+ * @returns {Promise}
+ */
+function existsIndex(index) {
+  return client.indices.exists({
+    index: index
+  });
+}
+
+/**
+ * Indices should have a different name than the alias
+ * Append the `aliasSuffix` to newly created indices
+ *
+ * @param {string} alias
+ * @returns {string}
+ */
+function createIndexName(alias) {
+  return alias + aliasSuffix;
+}
+
+/**
+ * Delete an Elasticsearch index
+ *
+ * @param {string} index
+ * @returns {Promise}
+ */
+function delIndex(index) {
+  return client.indices.delete({
+    index: index
+  }).then(function () {
+    log('info', 'Successfully deleted ' + index);
+  })
+    .catch(function (error) {
+      log('error', error);
+      return bluebird.reject(error);
+    });
+}
```

Comment:
Not entirely sure, but shouldn't it be available just in case anyone needed it? I figured that since it's open source someone might want it eventually.